lib/src/errors.ts

```diff
@@ -27,91 +31,84 @@ export class TdfError extends Error {
   }
 }
 
-export class UnsafeUrlError extends Error {
-  override name = 'UnsafeUrlError';
-  readonly url: string;
 
-  constructor(message: string, url: string) {
-    super(message);
-    Object.setPrototypeOf(this, new.target.prototype);
-    this.url = url;
-  }
+/**
+ * Errors that indicate the client or method does not have valid options.
+ */
+export class ConfigurationError extends TdfError {
+  override name = 'ConfigurationError';
 }
 
-export class AttributeValidationError extends TdfError {
+/**
+ * The assigned data attribute is not in the correct form.
+ */
+export class AttributeValidationError extends ConfigurationError {
   override name = 'AttributeValidationError';
+  attribute: unknown;
+  constructor(message: string, attribute: unknown, cause?: Error) {
+    super(message, cause);
+    this.attribute = attribute;
+  }
 }
 
-export class KasDecryptError extends TdfError {
-  override name = 'KasDecryptError';
-}
-
-export class KasUpsertError extends TdfError {
-  override name = 'KasUpsertError';
-}
-
-export class KeyAccessError extends TdfError {
-  override name = 'KeyAccessError';
-}
-
-export class KeySyncError extends TdfError {
-  override name = 'KeySyncError';
-}
-
-export class IllegalArgumentError extends Error {}
-
-export class IllegalEnvError extends Error {}
-
-export class InvalidCipherError extends TdfError {
-  override name = 'InvalidCipherError';
-}
-
-export class InvalidCurveNameError extends TdfError {
-  override name = 'InvalidCurveNameError';
-}
+/**
+ * Errors that indicate the TDF object is corrupt, invalid, or fails validation or decrypt.
+ */
+export class InvalidFileError extends TdfError {}
 
-export class InvalidDataTypeError extends TdfError {
-  override name = 'InvalidDataTypeError';
+/**
+ * Indicates a decrypt failure, either due to an incorrect key, corrupt ciphertext, or inappropriate key parameters.
+ */
+export class DecryptError extends InvalidFileError {
+  override name = 'DecryptError';
 }
 
-export class InvalidEphemeralKeyError extends TdfError {
-  override name = 'InvalidEphemeralKeyError';
+export class IntegrityError extends InvalidFileError {
+  override name = 'IntegrityError';
```

Comment:
is this the equivalent of tamper?

Reply:
This is for either `ErrSegSigValidation` or `ErrRootSigValidation` in your PR. `InvalidFileError` is equivalent to `ErrTampered`.